pkg/chunk/cache/fifo_cache.go

```diff
@@ -240,3 +274,32 @@ func (c *FifoCache) Get(ctx context.Context, key string) (interface{}, bool) {
 	c.totalMisses.Inc()
 	return nil, false
 }
+
+func sizeOf(i interface{}) int {
+	switch v := i.(type) {
+	case string:
+		return len(v)
+	case []int8:
+		return len(v)
+	case []uint8:
+		return len(v)
+	case []int32:
+		return len(v) * 4
+	case []uint32:
+		return len(v) * 4
+	case []float32:
+		return len(v) * 4
+	case []int:
+		return len(v) * 8
+	case []uint:
+		return len(v) * 8
```

Comment:
FYI, these are machine dependent. It's not safe to assume it's `8` bytes, even if `8` is the most common case (ie. we don't release 32bit binaries and probably any production system runs on 64 bit). That being said, if we keep it like this I would add a comment to explain it.